packages/react-graphql/src/hooks/tests/useMutation.test.tsx

```diff
@@ -0,0 +1,197 @@
+import * as React from 'react';
+import ApolloClient from 'apollo-client';
+import {createGraphQLFactory} from '@shopify/graphql-testing';
+
+import {
+  mountWithGraphQL,
+  runPendingAsyncReactTasks,
+  prepareAsyncReactTasks,
+  teardownAsyncReactTasks,
+} from './utilities';
+import updatePetMutation from './fixtures/UpdatePetMutation.graphql';
+import useMutation from '../useMutation';
+
+const createGraphQL = createGraphQLFactory();
+
+describe('useMutation', () => {
+  beforeEach(() => {
+    prepareAsyncReactTasks();
+  });
+
+  afterEach(() => {
+    teardownAsyncReactTasks();
+  });
+
+  describe('default', () => {
+    const MockMutation = ({children}) => {
```

Comment:
It is actually really difficult to test `useMutation` by itself because all the react hook doesn't work outside of a component. I do believe we have a doc that talks about it more.

https://github.com/Shopify/web-foundation/blob/ca2d34ff491eca187f9627a261ee0e784819854c/Best%20practices/React/Testing.md#hooks